tests/ui/test_ui.py

```diff
@@ -210,13 +213,15 @@ def test_show_right_panel(self, mocker, view,
         else:
             view.body.options.assert_not_called()
 
-    def test_keypress_normal_mode_navigation(self, view, mocker,
+    def test_keypress_normal_mode_navigation(self, view, mocker, widget_size,
                                              navigation_key_expected_key_pair):
         key, expected_key = navigation_key_expected_key_pair
         view.users_view = mocker.Mock()
         view.body = mocker.Mock()
         view.user_search = mocker.Mock()
-        size = (20,)
+        size = widget_size(view.main_window)
```

Comment:
Why does this size depend on `main_window` when we're calling the test on `view`?
(similarly elsewhere)

Reply:
The `View` class is a `WidgetWrap` subclass, which means its keypress size depends on the main widget passed in its `init` call.

As the `widget_size` uses the widget instance to deduce the widget type and ultimately the `size` tuple, the actual widget, which happens to be the one that we return from the `main_window`, is being passed.

```python3
class View(urwid.WidgetWrap):
        ...
        super().__init__(self.main_window())
```
```python3
    def main_window(self) -> Any:
        ...
        w = urwid.Frame(self.body, title_bar, focus_part='body',
                        footer=self.footer_view())
        return w
```

Reply:
From what I can tell, a `WidgetWrap` subclass acquires at least the minimal widget interface of the passed-in widget by default, including sizing? I assume we patch `main_window` to avoid constructing the UI; can we simply then patch `View.sizing` to provide what `super().__init__` would do for us, or something like that? We should add comments as to why we're doing that, similar to what you have now, but that keeps the tests themselves clearer?

It's still not ideal in that we're explicitly fixing the sizing of the widget in a mock, but I'm not sure we have an alternative given the current code layout? If we later changed the implementation however, the sizing should still come from the view - eg. if we were to inherit from `Frame` instead of `WidgetWrap` then the sizing would come from the view and the mock could be removed - but the tests would work the same.

```python3
>>> class X(urwid.WidgetWrap):
...     def __init__(self):
...             super().__init__(urwid.Text("hi"))
... 
>>> x = X()
>>> x
<X flow widget>
>>> dir(x)
['_DelegateToWidgetMixin__super', '_WidgetWrap__super', '_Widget__super', '_X__super', '__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_command_map', '_emit', '_invalidate', '_not_a_container', '_raise_old_name_error', '_repr_attrs', '_repr_words', '_selectable', '_set_w', '_sizing', '_w', '_wrapped_widget', 'base_widget', 'focus', 'focus_position', 'get_cursor_coords', 'get_pref_col', 'keypress', 'mouse_event', 'move_cursor_to_coords', 'pack', 'render', 'rows', 'selectable', 'sizing', 'w']
>>> x.sizing()
frozenset({'flow'})

Reply:
Thanks for the exposition; patching `View.sizing` seems like a better approach given the current implementation and how it makes the tests clearer and easier to change later.